addons/product/views/product_views.xml

```diff
@@ -233,7 +233,7 @@
                     <field name="product_tmpl_id" string="Product Template"/>
                 </field>
                 <filter name="categ_id" position="after">
-                    <filter string="Product Template" name="product_tmpl_id" context="{'group_by':'product_tmpl_id'}"/>
+                    <filter string="Product Template" name="tmpl_id" context="{'group_by':'product_tmpl_id'}"/>
```

Comment:
For more clarity, maybe keep the complete field name but add the `group_by` prefix
```suggestion
                    <filter string="Product Template" name="group_by_product_tmpl_id" context="{'group_by':'product_tmpl_id'}"/>
```

Reply:
Thanks, updated! :+1: